components/homme/src/share/cxx/mpi/Connectivity.hpp

```diff
@@ -78,23 +90,15 @@ class Connectivity
   //@name Getters
   //@{
 
-  // Get the view with all connections
-  template<typename MemSpace>
-  KOKKOS_INLINE_FUNCTION
-  typename std::enable_if<std::is_same<MemSpace,HostMemSpace>::value,
-                 HostViewUnmanaged<const ConnectionInfo*[NUM_CONNECTIONS]>>::type
-  get_connections () const { return h_connections; }
-
-  template<typename MemSpace>
-  KOKKOS_INLINE_FUNCTION
-  typename std::enable_if<std::is_same<MemSpace,ExecMemSpace>::value && !std::is_same<ExecMemSpace,HostMemSpace>::value,
-                 ExecViewUnmanaged<const ConnectionInfo*[NUM_CONNECTIONS]>>::type
-  get_connections () const { return m_connections; }
-
-  // Get a particular connection
-  template<typename MemSpace>
-  KOKKOS_INLINE_FUNCTION
-  const ConnectionInfo& get_connection (const int ie, const int iconn) const { return get_connections<MemSpace>()(ie,iconn); }
+  // Unstructured connections to handle RRM case. Connections for an element
+  // having local ID ie are
+  //   ucon(ucon_ptr(ie)):ucon(ucon_ptr(ie+1)-1).
+  // Device d_ucon(i) is 1-1 with host h_ucon(i) but has a different, ~8x
```

Comment:
Not that it matters, but for the sake of not confusing the reader, I think the ratio is more like 2x (I count 12B vs 27B). But I think `sharing_local_remote_iconn` can also be a uint8_t, no?, making the ratio an exact 3x.

Reply:
Oh, thanks. Yes, 8x is outdated; I'll update it to "~2.5x" based on the following. Depending on padding and with my switch to uint8_t for several fields, the old one is between 27 and 32 and the new one is 12. (Before the uint8_t switch it was 40B.) The "8x" is from when I had 32 (with padding) vs 8, but then I decided it was better to store sharing_local_remote_iconn rather than compute it online. I use int for sharing_local_remote_iconn because I store the full offset to get a fully no-calc lookup in the pack kernels, figuring padding would end up making the struct at least 10 and possibly 12B anyway.